numpy/core/tests/test_multiarray.py

```diff
@@ -342,6 +342,10 @@ def test_array_cont(self):
         assert_(np.ascontiguousarray(d).flags.c_contiguous)
         assert_(np.asfortranarray(d).flags.f_contiguous)
 
+    def test_npy_char_deprecation(self):
```

Comment:
Its not really important to use that infrastructure anymore I think. That said, this may be correct, since the test checks both that the warning is given, as well as tests that an error is raised when the warning is set to "raise".